pkg/snmp/traps/oid_resolver.go

```diff
@@ -163,12 +183,45 @@ func (or *MultiFilesOIDResolver) updateFromReader(reader io.Reader, unmarshalMet
 
 func (or *MultiFilesOIDResolver) updateResolverWithData(trapDB trapDBFileContent) {
 	definedVariables := variableSpec{}
-	for variableOID, variableData := range trapDB.Variables {
-		variableOID := NormalizeOID(variableOID)
+
+	allOIDs := make([]string, 0, len(trapDB.Variables))
+	for variableOID := range trapDB.Variables {
+		if !IsValidOID(variableOID) {
+			log.Errorf("trap variable OID %s does not look like a valid OID", variableOID)
+			continue
+		}
+		allOIDs = append(allOIDs, NormalizeOID(variableOID))
+	}
+
+	// "Fast" algorithm used to mark OID that act both as a variable and as a parent of other variable
+	// with 'isNode: true'. i.e if an OID <FOO>.<BAR> exists in the trapsDB but <FOO> also exists in the trapsDB
+	// then <FOO> acts as a 'Node' of the OID tree and should not be considered a match for resolving variables.
+	// In this fast algorithm the list is sorted then each OID is compared with its successor. It the successor starts
+	// with the current OID + a dot, then the current OID is a Node. 'Dots' are before digits in the lexicographic order.
+	// Note that in practice, OIDs that act both as Node and Leaf of the OID tree is extremely rare and is not expected
+	// in normal circumstamces. Thing is they sometimes exist.
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Clever way to solve the issue using domain specificities (lexicographical order or dot and digits). Not sure if I like it or not, but interesting alternative to Trie.